### ISSUE TITLE
Measuring performance of one user interaction

### ISSUE THREAD

**Issue by usherfu** [sitespeedio/browsertime#194]
Hi,
As mentioned in [issue reported in sitespeed.io](https://github.com/sitespeedio/sitespeed.io/issues/1224), I am trying to find way to measure performance of single step out of multiple steps. The following example scenario below is to navigate to https://www.google.com, type some keyword, and then click on go button. And I'd like to measure only the step of "click GO button". 

The idea is to put all steps prior to  "click Go button" in "preScript" section, and "click GO button" as part of the selenium script to be measured. 

The script would look like something like below:

```
function(browser) {
    return browser.by.cssSelector("button[name='btnG']")
    .then(function(el) {
            return el.click();
    }).catch(function(err) {
                    console.log(err);
            });
}],
```

 preScript:

```
module.exports = {
  run(context) {
    context.log.info('In pretask!!!');
    if (!context.taskData.loadedSitespeed) {
      return context.runWithDriver((driver) => {
          return driver.get('https://www.google.com')
            .then(() => browser.By.cssSelector("#lst-ib"))
            .then((el) => {
              el.type("performance");
            });
        })
        .then(() => {
          context.taskData.loadedSitespeed = true;
        });
    }
  }
};
```

With this approach, will the returned result of browsertime be the measurement of only "click Go button" step? Or am I misunderstanding how the browsertime works in general?

**Comment by soulgalore** [comment]
Hey @usherfu , it will not work out of the box. we need to fix so that you can execute selenium code in the measurement step or add something to start collecting metrics. I haven't discussed with @tobli how we should do it.

**Comment by usherfu** [comment]
Hey @soulgalore, thanks for the feedback. Looking forward to see how this can be supported in 4.0! Thanks!

**Comment by magnusgf** [comment]
I'm curious, is something like this possible now?

**Comment by soulgalore** [comment]
@magnusgf no. with the new 3.0 the code makes it easier but we haven't implemented it. we are not sure how it would work. It's been discussed a little here: https://github.com/sitespeedio/sitespeed.io/issues/1827 but been on ice for a long time.

**Comment by soulgalore** [comment]
This is implemented in Browsertime 4.0, look at the scripting capability.
